appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/AppSecKitInitializer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * -
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full license.
+ */
+
+package com.vaadin.appsec.v7.service;
+
+import javax.servlet.annotation.WebListener;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.appsec.backend.AppSecService;
+import com.vaadin.server.VaadinService;
+
+@WebListener
+public class AppSecKitInitializer implements HttpSessionListener {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(AppSecKitInitializer.class);
+
+    @Override
+    public void sessionCreated(HttpSessionEvent se) {
+        VaadinService vaadinService = VaadinService.getCurrent();
+        if (vaadinService != null) {
```

Comment:
Should we add here some kind of log message if for some reason the `vaadinService` is null and we can't init the AppSec Kit?

Reply:
Good point, I will add that. This can actually happen if a vaadin app is integrated as a part of e.g. JPS/JSF application - then the session init would be done by JSP/JSF and not vaadin thus VaadinService.getCurrent() being null. We can't really address this case in V7 version but logging an error should give the user an opportunity to address this.